### ISSUE TITLE
Support password authentication in the Solr plugin

### ISSUE THREAD

**Issue by frizner** [influxdata/telegraf#5829]
## Feature Request
Support password authentication in the Solr plugin, using Basic HTTP Authentication.
### Proposal:
The Solr plugin should work with Solr instances with enabled authentication.
### Current behavior:
401 status code when the plugin tries to request Solr node with enabled authentication.
### Desired behavior:
200 status code
### Use case: 
Solr nodes, especially in prod environment, often use the authentication.

**Comment by danielnelson** [comment]
Agreed, we should include the regular HTTP options along with this:
```
  ## Optional HTTP Basic Auth Credentials
  # username = "username"
  # password = "pa$$word"

  ## Optional TLS Config
  # tls_ca = "/etc/telegraf/ca.pem"
  # tls_cert = "/etc/telegraf/cert.pem"
  # tls_key = "/etc/telegraf/key.pem"
  ## Use TLS but skip chain & host verification
  # insecure_skip_verify = false

  ## Amount of time allowed to complete the HTTP request
  # timeout = "5s"
```

**Comment by frizner** [comment]
I've changed solr.go in order to have authenticated access and tested it with my Solr instances. May I push a request to some  branch?

**Comment by danielnelson** [comment]
Yes, create a fork of this repo and then push it to a branch on the fork, then you can create the pull request from there.

**Comment by frizner** [comment]
#5832

**Comment by frizner** [comment]
#5832

**Comment by danielnelson** [comment]
Closed in #5832
